library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/Base64Serializer.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.mapper
+
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.util.DisplayMetrics
+import androidx.annotation.MainThread
+import com.datadog.android.sessionreplay.internal.async.RecordedDataQueueRefs
+import com.datadog.android.sessionreplay.internal.utils.Base64Utils
+import com.datadog.android.sessionreplay.internal.utils.DrawableUtils
+import com.datadog.android.sessionreplay.model.MobileSegment
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+private val THREAD_POOL_MAX_KEEP_ALIVE_MS = TimeUnit.SECONDS.toMillis(5)
+private const val CORE_DEFAULT_POOL_SIZE = 1 // Only one thread will be kept alive
+
+class Base64Serializer internal constructor(
+    private val threadPoolExecutor: ThreadPoolExecutor = ThreadPoolExecutor(
```

Comment:
ideally this executor should have only one instance and injected into this class. This way you could have many instances of this class but share the executor. 
There is another possibility where you could keep it like this but then you make sure that you only have a single instance of it and injected into the constructor of all the Mappers. The way I see it the first option is easier.

Reply:
I've moved it in the meantime to an external singleton in order to avoid modifying the signatures of the mappers. If we find a better way to do this we can modify it later on